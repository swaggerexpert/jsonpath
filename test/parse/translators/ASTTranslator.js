import { assert } from 'chai';
import { jestExpect as expect } from 'mocha-expect-snapshot';

import { parse, ASTTranslator } from '../../../src/index.js';

describe('parse', function () {
  context('translators', function () {
    context('ASTTranslator', function () {
      specify('should translate a JSONPath to an AST', function () {
        const parseResult = parse('$["a"]', { translator: new ASTTranslator() });

        assert.isTrue(parseResult.result.success);
        expect(parseResult.tree).toMatchSnapshot();
      });
    });
  });
});
