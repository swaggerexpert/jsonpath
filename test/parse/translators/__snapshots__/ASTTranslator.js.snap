// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`parse translators ASTTranslator should translate a JSONPath to an AST 1`] = `
{
  "type": "JsonPathQuery",
  "segments": [
    {
      "type": "ChildSegment",
      "selector": {
        "type": "BracketedSelection",
        "selectors": [
          {
            "type": "NameSelector",
            "value": "a",
            "format": "double-quoted",
          },
        ],
      },
    },
  ],
}
`;
